### ISSUE TITLE
React Grid Column Reordering broken in 3.0.0?

### ISSUE THREAD

**Issue by tyddynonn** [DevExpress/devextreme-reactive#3509]
Moved @devexpress/dx-react-grid to 3.0.0 from 2.76, also with @devexpress/dx-react-grid-bootstrap4

Snip of Grid config is:
```                            <DragDropProvider />                                
                                <Table
                                    key='FlightID'
                                    tableComponent={StripedTable}
                                />
                                <TableColumnReordering
                                    order={columnorder}
                                    onOrderChange={(colorder:string[]) =>{
                                        setColumnorder(colorder);
                                        localStorage.setItem(FLIGHTS_COLUMNORDER, JSON.stringify(colorder));
                                    }}  
                                /> 
```
(there is separate code to set columnorder on page load)

With 2.7.6, dragging columns to reorder works and is persisted to localStorage as expected.

with 3.0.0, columns appear to drag, but onOrderChange event is not fired on drop, *except* on first load of grid.

So, one reordering action can be done but none thereafter.

Reverting back to 2.7.6 resolves the issue without any code changes

React 17, Chrome

**Comment by Krijovnick** [comment]
Hi @tyddynonn,

I created examples with column reordering in v[2.7.6](https://codesandbox.io/s/epic-microservice-o5n9n?file=/index.js) and v[3.0.0](https://codesandbox.io/s/affectionate-jackson-cluj0?file=/demo.js). In both versions, the `onOrderChange` event fires in the same way. Could you modify the examples to show when the event works differently.

**Comment by tyddynonn** [comment]
It's a weird one. If I modify your 3.0.0 examples to use the useEffect/persistence logic I have in my app, column reordering still works.
There must be some kind of side effect from logic elsewhere in my app. I will dig deeper and post back,.

**Comment by tyddynonn** [comment]
My app fetches the row data from an API, using server-side sorting and paging. 
Anything that triggers a data fetch (e.g. a sort or a page switch) 'resets' the issue with column ordering and allows a new successful drag and drop operation.
When time allows I will try to create a sandbox that uses the same architecture as my app & see if that reproduces the problem

**Comment by tyddynonn** [comment]
I finally got to the bottom of this. 

I was using a dedicated component to render the table:

```
export const StripedTable: React.FC<TableProps> = (props) => (
    <Table.Table
        {...props}
        className="table-striped"
    />
);
```

and in my Grid definition

```
<Table
   columnExtensions={columnwidths}
   tableComponent={StripedTable}
/>
```

In 2.7.6, this all worked fine. In 3.0.0 (and 3.0.1) dragging columns is broken. 

Removing  
```  tableComponent={StripedTable} ``` fixed the problem in 3.0.0 and 3.0.1. 

No idea why.

**Comment by Krijovnick** [comment]
@tyddynonn 

I didn't manage to reproduce the issue. Could you please modify this [example](https://codesandbox.io/s/goofy-austin-vbghbq?file=/demo.js) to demonstrate the problematic behavior?
